src/web_app/js/windowport.js

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+/* More information about these options at jshint.com/docs/options */
+
+/* globals trace, chrome */
+/* exported WindowPort */
+
+'use strict';
+
+// This is used to communicate from the Chrome App window to background.js.
+// It opens a Port object to send and receive messages. When the Chrome
+// App window is closed, background.js receives notification and can
+// handle clean up tasks.
+var WindowPort = (function() {
```

Comment:
I was indeed trying to add a namespace, as we have so many globals already, it seems desirable to at least slow the rate of increase. I'll change it if you want though.

The difference between what I wrote and your first example is only the namespace (as lazy initialization of the port object) as far as I can tell.

For the second example, the actual port object (returned by connect()) should be a singleton. The reason is that on the background side, the port object is what ties the requests to their correct web socket, so the cleanup queue requests need to come to the same port to be able to use the web socket created by RemoteWebSocket's requests.